quay/src/v1/mod.rs

```diff
@@ -0,0 +1,106 @@
+use failure::Fallible;
+use reqwest::{self, async};
+
+mod manifest;
+
+mod tag;
+pub use self::tag::Tag;
+
+static DEFAULT_API_BASE: &str = "https://quay.io/api/v1/";
+
+/// Client to make outgoing API requests to a quay instance.
+#[derive(Clone, Debug)]
+pub struct Client {
+    /// Base URL for API endpoint.
+    api_base: reqwest::Url,
+    /// Asynchronous reqwest client.
+    hclient: async::Client,
+    /// Authentication token.
+    token: Option<String>,
+}
+
+impl Client {
+    /// Return a client builder with default options.
+    pub fn builder() -> ClientBuilder {
+        ClientBuilder::default()
+    }
+
+    /// Return a request builder with base URL and parameters set.
+    pub(crate) fn new_request<S: AsRef<str>>(
+        &self,
+        method: reqwest::Method,
+        url_suffix: S,
+    ) -> Fallible<async::RequestBuilder> {
+        let url = self.api_base.clone().join(url_suffix.as_ref())?;
+        let builder = {
+            let plain = self.hclient.request(method, url);
+            match self.token {
+                None => plain,
+                Some(ref token) => {
+                    let bearer_token = format!("Bearer {}", token);
+                    plain.header("Authorization", bearer_token)
+                }
+            }
+        };
+        Ok(builder)
+    }
+}
+
+/// Client to make outgoing API requests to a quay instance.
+#[derive(Clone, Debug)]
+pub struct ClientBuilder {
+    api_base: Option<String>,
+    hclient: Option<async::Client>,
+    token: Option<String>,
+}
+
+impl ClientBuilder {
+    /// Set (or reset) the HTTP client to use.
+    pub fn http_client(self, hclient: Option<async::Client>) -> Self {
```

Comment:
I would add a `set_` or `with_` prefix to this method name because it usually means to get the thing if it doesn't say otherwise.

Reply:
The advice is generally true. However in this specific case this a builder, and in my experience so far builders don't explicitly label methods as setters (I guess because they don't have getters):
 * https://doc.rust-lang.org/std/fs/struct.DirBuilder.html
 * https://doc.rust-lang.org/std/fs/struct.OpenOptions.html
 * https://docs.rs/reqwest/0.9.9/reqwest/struct.ClientBuilder.html

I'll leave these methods without the prefix (same for the following comments too).